src/sagas/nanoContract.js

```diff
@@ -75,14 +92,169 @@ export function* registerNanoContract({ payload }) {
     blueprintId: ncState.blueprint_id,
     blueprintName: ncState.blueprint_name
   };
-  yield call(wallet.storage.registerNanoContract, ncId, nc);
+  yield call(wallet.storage.registerNanoContract.bind(wallet.storage), ncId, nc);
 
+  log.debug('Success registering Nano Contract.');
   // emit action NANOCONTRACT_REGISTER_SUCCESS
   yield put(nanoContractRegisterSuccess({ entryKey: ncId, entryValue: nc }));
 }
 
+/**
+ * @typedef {Object} RawNcTxHistory
+ * @property {string} hash
+ * @property {Object[]} inputs
+ * @property {Object[]} outputs
+ * @property {string} nc_args
+ * @property {string} nc_blueprint_id
+ * @property {string} nc_id
+ * @property {string} nc_method
+ * @property {string} nc_pubkey
+ * @property {number} nonce
+ * @property {Object[]} parents
+ * @property {number} signal_bits
+ * @property {Date} timestamp
+ * @property {string[]} tokens
+ * @property {number} version
+ * @property {number} weight
+ * @property {boolean} is_voided
+ *
+ * @typedef {Object} RawNcTxHistoryResponse
+ * @property {boolean} success
+ * @property {number|null} after
+ * @property {RawNcTxHistory} history
+ */
+
+/**
+ * @typedef {Object} NcTxHistory - Nano Contract's Transactions History
+ * @property {string} txId - transaction hash
+ * @property {number} timestamp
+ * @property {string} tokens - list of token's uid
+ * @property {{[uid: string]: Object}} balance - balance per token
+ * @property {boolean} isVoided - flag for transaction void
+ * @property {string} ncId - ID of Nano Contract
+ * @property {string} ncMethod - method called on Nano Contract
+ * @property {string} blueprintId - id of the Blueprint instantiated by Nano Contract
+ * @property {string} caller - address of the caller
+ * @property {boolean} isMine - flag indicating the caller address belongs to this wallet
+ */
+
+/**
+ * Fetch history from Nano Contract wallet's API.
+ * @param {string} ncId Nano Contract ID
+ * @param {number} count Maximum quantity of history items
+ * @param {string} after Transaction hash to start to get items
+ * @param {Object} wallet Wallet instance from redux state
+ */
+export async function fetchHistory(ncId, count, after, wallet) {
+  /**
+   * @type {RawNcTxHistoryResponse} response
+   */
+  const response = await ncApi.getNanoContractHistory(ncId, count, after);
```

Comment:
suggestion: add `@throws` to the jsdocs explaining that it will throw if the request returns a status code < 399 or if `success` is false so we know that this method does not attempt to handle errors

Reply:
Added.

Reply:
@alexruzenhack It's `throws` in jsdocs (https://jsdoc.app/tags-throws). This might be an issue for LSPs